File: server/java/render/src/main/java/com/metaobjects/render/recover/FieldRecovery.java
package com.metaobjects.render.recover;

/** FROZEN cross-port per-field recovery classification. Do not reorder or add without an ADR. */
public enum FieldRecovery {
    RECOVERED,
    // FR-011: a value reached via @coerceDefault (present-but-uncoercible fallback) or
    // @default (absent-fill). Classified by Recover via the terminal coercion kind.
    DEFAULTED,
    LOST_OPTIONAL,
    LOST_REQUIRED,
    MALFORMED
}

File: server/csharp/MetaObjects.Render/Recover/FieldRecovery.cs
namespace MetaObjects.Render.Recover;

/// <summary>
/// FROZEN cross-port per-field recovery classification. Names match the conformance corpus
/// expected.json exactly — the runner uses <c>.ToString()</c> comparison.
/// Do not reorder or rename without an ADR.
/// </summary>
public enum FieldRecovery
{
    RECOVERED,
    // DEFAULTED is reserved (a future @default-backed value); the current engine does not emit it.
    DEFAULTED,
    LOST_OPTIONAL,
    LOST_REQUIRED,
    MALFORMED,
}
